src/lib/storage/up-storage.js

```diff
@@ -10,19 +10,37 @@ const Logger = require('../logger');
 const MyStreams = require('@verdaccio/streams');
 const Utils = require('../utils');
 const zlib = require('zlib');
+const userAgentParser = require('../utils/npm-user-agent-parser');
 
 const encode = function(thing) {
   return encodeURIComponent(thing).replace(/^%40/, '@');
 };
 
-const jsonContentType = 'application/json';
-
-const contenTypeAccept = [
+const legacyAcceptContentType = 'application/json';
+const acceptContentType = [
   'application/octet-stream',
   'application/vnd.npm.install-v1+json; q=1.0',
-  `${jsonContentType} q=0.8, */*`,
+  `application/json q=0.8, */*`,
 ].join(', ');
 
+const determineContentType = (req) => {
+  if (!req || !req.headers) {
```

Comment:
Please, us `lodash` in order to avoid future `falsy` values. `_.isNil(foo) == false` etc etc.